query-builder/src/main/java/com/datastax/oss/driver/api/querybuilder/BuildableQuery.java

```diff
@@ -47,14 +47,32 @@
    * SimpleStatement.newInstance(asCql())
    * }</pre>
    *
-   * In addition, some query implementation might try to infer additional statement properties (such
-   * as {@link Statement#isIdempotent()}).
+   * In addition, some implementations might try to infer additional statement properties (such as
+   * {@link Statement#isIdempotent()}).
    */
   @NonNull
   default SimpleStatement build() {
     return SimpleStatement.newInstance(asCql());
   }
 
+  /**
+   * Builds the CQL query and wraps it in a simple statement, also providing values for bind
+   * markers.
+   *
+   * <p>This is a similar to:
+   *
+   * <pre>{@code
+   * SimpleStatement.newInstance(asCql(), values)
+   * }</pre>
+   *
+   * In addition, some implementations might try to infer additional statement properties (such as
+   * {@link Statement#isIdempotent()}).
+   */
+  @NonNull
+  default SimpleStatement build(Object... values) {
```

Comment:
While at it, why not also add variants for named values?

Reply:
Indeed. `SimpleStatement.newInstance` has that variant, so it makes sense to add it here too. Will do.